### ISSUE TITLE
Multiple word searching with List.search

### ISSUE THREAD

**Issue by sheffieldnick** [javve/list.js#680]
At present `List.search` matches list items against a simple substring, and while `List.fuzzySearch` supports multiple words, it is very slow and seems to be better suited to fuzzy matching larger blocks of text rather than many short strings?

Below is an extension of my faster search function in #679 that now supports multiple-word searches (space-separated words, logical AND) as well as quoted phrases that are matched exactly:
```javascript
listObj.search('Jon 1986', searchFunction); // Custom search for Jon AND 1986
listObj.search('"Jonny S" 1986', searchFunction); // Custom search for "Jonny S" AND 1986
```
note that whitespace is stripped from search words unless contained inside quotes. So `search('" a "')` (matches the separate word 'a') is very different to `search(' a ')` (matches anything with the character 'a').

This search function is faster than the default `List.search` in almost all cases, even searching on a single word, and is always _many_ times faster than `List.fuzzySearch`.
```javascript
// searchString is converted to lowercase by List.js
function searchFunction(searchString, columns){
  // Extract quoted phrases "word1 word2" from original searchString
  var words = [],  phrase;
  while ((phrase = searchString.match(/"([^"]+)"/)) !== null) {
    words.push(phrase[1]);
    searchString = searchString.substring(0,phrase.index) + searchString.substring(phrase.index+phrase[0].length);
  };
  // Get remaining space-separated words (if any)
  searchString = searchString.trim();
  if (searchString.length) words = words.concat(searchString.split(/\s+/));
  for (var k = 0, kl = listObj.items.length; k < kl; k++) {
    var item = listObj.items[k];
    item.found = false;
    if (!words.length) continue;
    var values = item.values();
    for (var i = 0, il = words.length; i < il; i++) {
      var word_found = false;
      for (var j = 0, jl = columns.length; j < jl; j++) {
        var column = columns[j];
        if (values.hasOwnProperty(column) && values[column] !== undefined && values[column] !== null) {
          var text = (typeof values[column] !== 'string') ? values[column].toString() : values[column];
          if (text.toLowerCase().indexOf(words[i]) !== -1) {
            // word found, so no need to check it against any other columns
            word_found = true;
            break;
          }
        }
      }
      // this word not found? no need to check any other words, the item cannot match
      if (!word_found) break;
    }
    item.found = word_found;
  }
};
```

**Comment by sheffieldnick** [comment]
Now updated to move `item.values()` call out of inner loops, thanks to @buczek

**Comment by javve** [comment]
Thank you so much! This is now merged into #696
